src/tree.js

```diff
@@ -70,6 +70,24 @@ export let reduceTree = (next = traverse) =>
     return result
   })
 
+let writeProperty = (node, index, [parent]) => {
+  parent[index] = node
```

Comment:
This is not structure preserving. Also, nested nodes will overwrite nodes at further up if the indexes are the same.